syntax = "proto3";


package hypernets.dispatchers.process.grpc.proto;


service ProcessBroker {
  rpc run(stream ProcessRequest) returns (stream DataChunk) {}
  rpc download(DownloadRequest) returns (stream DataChunk) {}
}

message ProcessRequest {
  string program = 1;
  repeated string args = 2;
  string cwd = 3;
  int32  buffer_size = 4;
  string encoding = 5;
}

message DownloadRequest {
  string peer = 1;
  string path = 2;
  int32  buffer_size = 3;
  string encoding = 4;
}

message DataChunk {
  enum DataKind {
    IN = 0;
    OUT = 1;
    ERR = 2;
    DATA = 10;
    END = 99;
    EXCEPTION = 400;
  }
  DataKind kind = 1;
  bytes data = 2;
}
